### PR TITLE
feat: Add new Kubernetes Secret backend

### PR DESCRIPTION
**Pull request by dennislapchenko**
### Description
Add a new backend: Kubernetes Secret.

**Fixes:** https://github.com/argoproj-labs/argocd-vault-plugin/issues/516

### Checklist
Please make sure that your PR fulfills the following requirements:
- [x] Reviewed the guidelines for contributing to this repository
- [x] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] Tests for the changes have been updated
- [x] Are you adding dependencies? If so, please run `go mod tidy -compat=1.17` to ensure only the minimum is pulled in.
- [x] Docs have been added / updated
- [ ] Optional. My organization is added to USERS.md.

### Type of Change

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### CODE REVIEW

File: docs/backends.md [hunk 1]
```diff
@@ -705,3 +705,66 @@ type: Opaque
...
+###### Inline Path
+
+```yaml
+kind: Secret
+apiVersion: v1
+metadata:
+  name: test-secret
+type: Opaque
+data:
+  password: <path:custom-secret-data#key>
```
Comment (werne2j) [inline docs/backends.md:769]: Can you add an example for an inline path pulling a secret from another namespace?

File: docs/backends.md [hunk 2]
```diff
@@ -705,3 +705,66 @@ type: Opaque
 data:
   password: <path:secret-id#key | base64encode>
 ```
+
+### Kubernetes Secret
+
+Setup the backend with default secret from which to conveniently inject values from.
```
Comment (werne2j) [inline docs/backends.md:711]: I don't love the idea of having to have a "default" secret as an env var. I think just specifying the kubernetes secret name in the manifests as inline or annotation is enough.

**Review by werne2j — COMMENTED** [review]
Thanks for the contribution @dennislapchenko! My biggest question is around the idea of the "default" secret and whether it is necessary.

**Comment by dennislapchenko** [comment]
hey @werne2j , any info on when this can be released? Or any other comments please?

**Comment by dennislapchenko** [comment]
Hi @werne2j or @jkayani , apologies for being so persistent.
How would you like to proceed with this? Is everything to your liking?

**Review by werne2j — APPROVED** [review]
Thanks @dennislapchenko!

**Comment by acelinkio** [comment]
Any chance we could cut a new release to close https://github.com/argoproj-labs/argocd-vault-plugin/issues/516 .  Super excited to use this!

**Comment by dennislapchenko** [comment]
> Any chance we could cut a new release to close https://github.com/argoproj-labs/argocd-vault-plugin/issues/516 .  Super excited to use this! 

Agree!

Maybe you can accelerate it if you submit a 1.16 release PR with version bumps (they did similar with 1.15)

**Pull request merged**